0000-modules/detailed-design/migration.md

```diff
@@ -0,0 +1,167 @@
+# Migration
+
+The changes proposed by this RFC are significant breaking changes to the
+language. To complete them entirely we will need to pass a checkpoint/epoch
+boundary as described in [RFC #2052][checkpoints]. However, we will begin with
+a multi-phase deprecation process to ease the transition.
+
+We will also introduce a tool that will not only automatically move a crate
+onto a new system, but also *improve clarity* of the user's code by
+intentionally drawing the distinction between the `export` and `pub`
+visibilities that today does not firmly exist.
+
+## Initial, compatible mode
+
+In the first release supporting this RFC, we will begin supporting all of the
+feature set of this RFC, without breakages to existing functionality. That is:
+
+* `--extern` and `--module` arguments will be automounted, and cargo will pass
+`--module` arguments in its expected way.
+* The `crate::` path prefix will be accepted.
+* The `export` visibility will be accepted.
+* The "vis path" re-export syntax will be accepted.
+
+However, existing semantics will not be broken, meaning:
+
+* Module-internal paths will be accepted from the crate root without the
+`crate::` prefix.
+* `pub` will continue to have its current semantics if a crate does not ever
+use the `export` visibility
+
+In other words, as soon as this RFC is implemented, users will be able to
+transition to the new system, but they will not be immediately *required* to.
+
+The only breakage during this phase arises if users have files with valid Rust
+module names inside of the directory than defines a crate, which is not a
+compilable module of that crate. Users are encouraged to fix that by renaming
+those files so that they will not be picked up by cargo, or else deleting them
+if they are not necessary. **Note** that this does not include overlapping
+crates which are both managed by cargo, as when cargo detects that it will
+instead pass no `--module` arguments.
+
+### Using `export` changes semantics
```

Comment:
I wonder if it's possible to use `crate` keyword (currently only used in `extern crate`) instead of `pub`. This would avoid issues caused by changing well established meaning of `pub` keyword. Say...

```rust
crate fn hello() {
    println!("Hello, world!");
}

crate struct Example {
    crate a: i32,
    b: i64,
}
```